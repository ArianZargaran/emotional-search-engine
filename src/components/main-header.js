import React, { useState } from "react";
import PropTypes from "prop-types";
import headerColorsArr from "utils/header-colors-array";

import "styles/main-header.css";

export function MainHeader({ children }) {    
  const [ colorIdx , setColorIdx ] = useState(0);
  const [ headerColor , setHeaderColor ] = useState(headerColorsArr[0]);

  const changeHeaderColor = () => {    
    setColorIdx(currentColorIdx => {
      return currentColorIdx === headerColorsArr.length - 1 ? 0 : currentColorIdx + 1
    });
    setHeaderColor(headerColorsArr[colorIdx]);
  };

  return ( 
    <h1
      style={{"color": headerColor}}
      className="ese-main-header"
      onMouseOver={changeHeaderColor}
    >
      {children}
    </h1>
  );
};

MainHeader.propTypes = {
  children: PropTypes.string
};

MainHeader.defaultProps = {
  children: "Here goes a title"
};

export default React.memo(MainHeader);