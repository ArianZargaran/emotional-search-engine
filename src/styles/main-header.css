.ese-main-header {
  font-weight: bold;
  font-size: 40px;
  margin: 30px auto;
  text-align: center;
  color: #4688F1;
}