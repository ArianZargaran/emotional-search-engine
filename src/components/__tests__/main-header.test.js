import React from 'react';
import { MainHeader } from "../main-header";
import headerColorsArr from "utils/header-colors-array";

const setup = (extraProps) => {
  const props = {
    children: "Here goes a title",
    ...extraProps
  };

  const component = shallow(<MainHeader />);

  return { props, component };
};

it('renders MainHeader correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<MainHeader />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

describe('MainHeader functional component', () => {
  it('Has proper className', () => {
    const { component } = setup();
    const elClass = component.find('.ese-main-header');
    expect(elClass).toHaveLength(1);
  });

  it("Has a default Children prop", () => {
    const { component } = setup();
    const elChildren = component.props().children;
    expect(elChildren).toBe("Here goes a title");
  });

  it("Has HeaderColorsArr[0] as its initial color", () => {
    const { component } = setup();
    let containerStyle = component.find('.ese-main-header').prop('style');
    expect(containerStyle).toEqual({color: headerColorsArr[0]});
  });

  //TODO: simulate hover effect and then check '.ese-main-header' color;
});