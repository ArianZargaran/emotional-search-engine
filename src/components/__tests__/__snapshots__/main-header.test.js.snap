// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainHeader correctly 1`] = `
<h1
  className="ese-main-header"
  onMouseOver={[Function]}
  style={
    Object {
      "color": "#4688F1",
    }
  }
>
  Here goes a title
</h1>
`;
